### PR TITLE
Marker annotations

### PR DESCRIPTION
**Pull request by jjmason**
Merging marker annotations back to active-directory.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,33 +62,78 @@ dedicated for this purpose.  You must also set the `CONJUR_APPLIANCE_URL` and
...
+### Active Directory Support
+
+The conjur-ldap-sync tool also supports Active Directory [user](https://msdn.microsoft.com/en-us/library/ms683980(v=VS.85).aspx)
+and [group](https://msdn.microsoft.com/en-us/library/ms682251(v=VS.85).aspx) objects in a similar 
+fashion if you provide the `--mode active_directory` flag.  This mode behaves differently from `posix`
+in several ways:
+    
+  * Users' logins are taken from the `cn` attribute instead of the `uid` attribute.
+  * Group memberships are determined by the `memberOf` attribute of user objects.
+  * Object's distinguished names are determined by the `distinguishedName` attribute
```
Comment (dividedmind) [inline README.md]: O_o AD is weird

File: README.md [hunk 2]
```diff
@@ -62,33 +62,78 @@ dedicated for this purpose.  You must also set the `CONJUR_APPLIANCE_URL` and
...
+ 
+### Custom Object Classes    
+
+The object classes selected for users and groups can be controlled by the `--group-object-classes`
+and `--user-object-classes` flags, whose defaults are determined by the `--mode` flag.
+
+### Source Tags
+
+In order to prevent subsequent runs of `conjur-ldap-sync` from colliding or altering users and groups that
+were created by Conjur and not from LDAP, the tool applies an [annotation](LINK) named `ldap-sync/source`
```
Comment (dividedmind) [inline README.md]: Link missing

File: README.md (hunk 1 above)
Comment (jjmason) [inline README.md]: IKR?

File: lib/conjur/ldap/roles.rb [hunk 3]
```diff
@@ -10,15 +9,29 @@ module Conjur::Ldap::Roles
...
   # 
-  # Because you can't delete roles Conjur, any roles that are``
+  # Because you can't delete roles Conjur, any roles that are
   # deleted in upstream are simply removed from all membership relations
-  # (except the admin one).
+  # that were created by the tool.
+  #
+  # Thus, if you add a user imported from LDAP to a group that was *not* imported
+  # from LDAP, the user will remain in that group.  This also prevents important roles
+  # such as the owner (or the role as which the ldap-sync tool runs).
```
Comment (dividedmind) [inline lib/conjur/ldap/roles.rb]: I something missing here.

File: lib/conjur/ldap/roles.rb [hunk 4]
```diff
@@ -30,37 +43,21 @@ module Conjur::Ldap::Roles
   # @option opts [Boolean] :import_ldap_ids (true) whether to import uids and gids from LDAP
```
Comment (dividedmind) [inline lib/conjur/ldap/roles.rb]: `opts` is no more, I believe

File: lib/conjur/ldap/roles.rb [hunk 5]
```diff
@@ -1,7 +1,6 @@
 # A mixin for Conjur::API implementing utility functions
```
Comment (dividedmind) [inline lib/conjur/ldap/roles.rb]: Not a mixin anymore, I believe.

File: lib/conjur/ldap/roles.rb (hunk 3 above)
Comment (jjmason) [inline lib/conjur/ldap/roles.rb]: This is old, I'll delete.

**Pull request merged**